### PR TITLE
Fix the default value of state for NestedEntropyTargetAlgorithm

### PR DESCRIPTION
**Pull request by emailweixu**
state is not used by NestedEntropyTargetAlgorithm and it is called without providing state in Agent. So it should have a default value for state.

### CODE REVIEW

File: alf/algorithms/algorithm.py [hunk 1]
```diff
@@ -1020,7 +1020,12 @@ def update_with_gradient(self,
 
         if isinstance(loss_info.loss, torch.Tensor):
             loss = weight * loss_info.loss
-            loss.backward()
+            # In theory, there might be unfinished kernel before backward is called
+            # and and after backward() returns and torch.cuda.synchronize() should
+            # be called to make sure we only measure the computation time for backward.
+            # In practice, we found that the measured time is almost unchanged.
```
Comment (hnyu) [inline alf/algorithms/algorithm.py]: Not sure what this paragraph means?

**Comment by hnyu** [comment]
EntropyTargetAlgorithm has no issue because its predict_step and rollout_step already have state=None. It is NestedEntropyTargetAlgorithm that this PR fixes.

File: alf/algorithms/algorithm.py (hunk 1 above)
Comment (emailweixu) [reply alf/algorithms/algorithm.py]: In general, there is no guarantee that its computation is finished when a torch function returns. See https://pytorch.org/docs/stable/notes/cuda.html#asynchronous-execution

**Comment by emailweixu** [comment]
> EntropyTargetAlgorithm has no issue because its predict_step and rollout_step already have state=None. It is NestedEntropyTargetAlgorithm that this PR fixes.

Changed PR title and description.

File: alf/algorithms/algorithm.py (hunk 1 above)
Comment (hnyu) [reply alf/algorithms/algorithm.py]: OK, I think it's a little difficult to parse the first sentence. Maybe 

"In theory, there might be unfinished kernel before backward being called, and after backward returns, torch.cuda.synchronize() should be called to make sure..."

**Review by hnyu — APPROVED** [review]

**Pull request merged**